### ISSUE TITLE
Quaternion[A].=== (and Complex) use cooperative equality

### ISSUE THREAD

**Issue by denisrosset** [non/spire#611]
See:
```scala

  def ===(that: Quaternion[_]): Boolean =
    r == that.r && i == that.i && j == that.j && k == that.k

  def =!=(that: Quaternion[_]): Boolean =
    !(this === that)
```
and
```scala
  def ===(that: Complex[_]): Boolean =
    real == that.real && imag == that.imag

  def =!=(that: Complex[_]): Boolean =
    !(this === that)
```

which is probably not the expected behavior when using `import spire.syntax.eq._`.

Or the behavior should be documented.

**Comment by denisrosset** [comment]
Addressed by #527 actually

**Comment by denisrosset** [comment]
Regrouped.
